test/e2e/specs/treeview/00-dynamic-treeview.spec.js

```diff
@@ -262,20 +276,282 @@ exports.tests = function (appName, baseUrl) {
+            });
+
+            it('node list expanded on + click', function() {
+              btn.click();
+              browser.wait(EC.visibilityOf(element(by.css('#nid_1434 > ul'))), browser.params.defaultTimeout);
+              element(by.css('#nid_1434')).all(by.tagName('li')).each(function (elm) {
+                expect((elm).isDisplayed()).toBeTruthy('Expanded List list is not visible.');
+              });
+            });
+
+            it('should open the image popup modal on click of image icon', function(done) {
+                browser.wait(EC.elementToBeClickable(imgLink), browser.params.defaultTimeout);
+                imgLink.click().then(function() {
+                    var modalContent = element(by.css('.modal-content'));
+                    deriva.waitForElement(modalContent);
+                    expect(modalContent.isDisplayed()).toBeTruthy();
+                    done();
+                }).catch(function (err) {
+                    done.fail();
+                    console.log(err);
+                });
+            });
+
+            it('should close the image popup modal on click of X on the modal window', function(done) {
+                 var closeBtn = element(by.css('#schematic-modal > div > div > div.modal-header > button'));
+                 var modalContent = element(by.id('schematic-modal'));
+                 browser.wait(EC.elementToBeClickable(closeBtn), browser.params.defaultTimeout);
+                 closeBtn.click().then(function(){
+                    deriva.waitForElementInverse(modalContent);
+                    expect(modalContent.isDisplayed()).toEqual(false);
+                    done();
+                }).catch(function (err) {
+                    console.log(err);
+                    done.fail();
+                });
+            });
+
+            it('should contain annotations', function(done) {
+                 var annotationsImgLink = expectedValue.annotations.imgs;
+                 imgFullLinks = annotationsImgLink.map(function (ele) {
+                   return baseUrl + ele;
+                 });
+
+                 element(by.id('nid_1431')).all(by.tagName('img')).getAttribute('src').then(function (linkArr) {
+                   linkArr.forEach(function (url) {
+                      expect(imgFullLinks).toContain(url);
+                      httpFun.httpGet(url).then(function (res) {
+                        expect(res.statusCode).toEqual(200, url + " is not a valid image source.");
+                        done();
+                      }).catch(function (err) {
+                        console.log(err);
+                        done.fail();
+                      });
+                    });
+                 });
+            });
+            it('density Note should have correct tooltip', function() {
+                var densityNote = element(by.css('#nid_1431_anchor > span > span:nth-child(3) > img:nth-child(4)'));
+                densityNote.click();
```

Comment:
Both places that you are clicking the density note icon, it _sometimes_ triggers before the modal dialog has finished fading out. I'm presented with this error:
`Failed: element click intercepted: Element <img src="resources/images/NerveDensity/note.gif" class="contains-note" title="" data-original-title="densityNote: Axons are scarce in the detrusor, where they are primarily on the dorsal aspect."> is not clickable at point (807, 401). Other element would receive the click: <div class="modal-backdrop fade"></div>`

You will have to play around with this a little bit, but I think you can use `elementToBeClickable` before clicking here. If that doesn't work, maybe checking that the modal element is no longer visible?

You could also try removing the modal animations so it just shows and dissappears rather than fading (this has helped with some of our other tests before).
```js
element(by.id("schematic-modal"))allowAnimations(false)
```